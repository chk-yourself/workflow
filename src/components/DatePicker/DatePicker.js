import React, { Component } from 'react';
import {
  toSimpleDateString,
  toSimpleDateObj,
  isSDSFormat,
  isSameDate
} from '../../utils/date';
import { Calendar } from '../Calendar';
import { Button } from '../Button';
import { Input } from '../Input';
import './DatePicker.scss';

export default class DatePicker extends Component {
  state = {
    today: {
      day: new Date().getDate(),
      month: new Date().getMonth(),
      year: new Date().getFullYear()
    },
    selectedDate: this.props.selectedDate,
    currentMonth: this.props.currentMonth,
    currentYear: this.props.currentYear,
    dateString: toSimpleDateString(this.props.selectedDate) || ''
  };

  resetCalendar = () => {
    this.setState({
      currentMonth: this.props.currentMonth,
      currentYear: this.props.currentYear
    });
  };

  componentDidMount() {}

  selectDate = date => {
    if (typeof date === 'string') {
      const newDate = toSimpleDateObj(date);
      const { month, year } = newDate;
      const { currentMonth, currentYear } = this.state;
      this.setState({
        selectedDate: newDate
      });
      if (month !== currentMonth || year !== currentYear) {
        this.setCurrentMonth({ month, year });
      }
    } else {
      this.setState({
        selectedDate: date,
        dateString: toSimpleDateString(date)
      });
    }
  };

  setCurrentMonth = ({ month, year }) => {
    this.setState({
      currentMonth: month,
      currentYear: year
    });
  };

  setCurrentYear = year => {
    this.setState({
      currentYear: year
    });
  };

  updateDateString = e => {
    const { value } = e.target;
    this.setState({
      dateString: value
    });
    if (isSDSFormat(value)) {
      this.selectDate(value);
    }
  };

  setDate = () => {
    const { selectedDate: currentDueDate, selectDate } = this.props;
    const { selectedDate } = this.state;
    if (
      !(currentDueDate === null && selectedDate === null) &&
      !isSameDate(currentDueDate, selectedDate)
    ) {
      const { day, month, year } = selectedDate;
      const date = new Date(year, month, day);
      selectDate(date);
    }
    this.closeDatePicker();
  };

  clearDueDate = () => {
    const { selectedDate, selectDate } = this.props;
    this.selectDate(null);
    if (selectedDate !== null) {
      selectDate(null);
    }
  };

  closeDatePicker = () => {
    const { onClose } = this.props;
    this.resetCalendar();
    onClose();
  }

  render() {
    const { isActive, innerRef } = this.props;
    const {
      today,
      selectedDate,
      currentMonth,
      currentYear,
      dateString
    } = this.state;
    return (
      <div
        className="date-picker"
        ref={innerRef}
        style={{ display: !isActive ? 'none' : 'block' }}
      >
        <div className="date-picker__header">
          <div className="date-picker__due-date-wrapper">
            <Input
              name="dueDate"
              type="text"
              label="Due Date"
              value={dateString}
              onChange={this.updateDateString}
              className="date-picker__input--due-date"
              labelClass="date-picker__label--due-date"
              placeholder="mm-dd-yy"
              maxLength={8}
            />
          </div>
        </div>
        <Calendar
          classes={{
            calendar: 'date-picker__calendar',
            weekDay: 'date-picker__week-day'
          }}
          month={currentMonth}
          year={currentYear}
          selectedDate={selectedDate}
          today={today}
          onDayClick={this.selectDate}
          onMonthClick={this.setCurrentMonth}
          onYearClick={this.setCurrentYear}
        />
        <div className="date-picker__footer">
          <Button
            size="sm"
            variant="text"
            color="neutral"
            className="date-picker__btn--clear"
            onClick={this.clearDueDate}
          >
            Clear
          </Button>
          <Button
            size="sm"
            variant="contained"
            color="primary"
            className="date-picker__btn--set-due-date"
            onClick={this.setDate}
          >
            Done
          </Button>
          <Button
            size="sm"
            variant="text"
            color="neutral"
            className="date-picker__btn--cancel"
            onClick={this.closeDatePicker}
          >
            Cancel
          </Button>
        </div>
      </div>
    );
  }
}
