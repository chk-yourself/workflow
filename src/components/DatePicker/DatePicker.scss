@import '../../styles/_variables';
@import '../../styles/_mixins';

.date-picker {
  background-color: #fff;
  border-radius: 4px;
  padding: 12px;
  @include box-shadow(3);
  position: absolute;
  z-index: 99;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}

.date-picker__btn--set-due-date {
  padding-top: 6px;
  padding-bottom: 6px;
}

.date-picker__btn--clear {
  padding-left: 0;
}

.date-picker__btn--cancel {
  padding-right: 0;
}

.date-picker__btn--set-due-date,
.date-picker__btn--clear,
.date-picker__btn--cancel {
  font-weight: 600;
}

.date-picker__calendar {
  width: 240px;
  margin: 0 auto;
}

.date-picker__week-day {
  height: 32px;
  line-height: 32px;
  font-weight: 700;
  color: $grey-800;
}

.date-picker__header {
  margin-bottom: 8px;
}

.date-picker__footer {
  margin-top: 8px;
  display: flex;
  justify-content: space-between;
  align-items: center;
}

.date-picker__label--due-date {
  font-size: 11px;
  text-align: center;
  font-weight: 500;
  color: $grey-700;
  margin-bottom: 4px;
}

.date-picker__due-date-wrapper {
  padding: 0 8px;
}

.date-picker__input--due-date {
  height: 30px;
  font-size: 14px;
  line-height: 1;
  padding: 0;
  display: block;
  width: 100%;
  text-align: center;
  margin: 0 auto;
  border-width: 1px;
}
