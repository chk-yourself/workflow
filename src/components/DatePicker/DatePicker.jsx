import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  toSimpleDateString,
  toSimpleDateObj,
  getSimpleDate,
  isSDSFormat,
  isSameDate
} from '../../utils/date';
import { Calendar } from '../Calendar';
import { Button } from '../Button';
import { Input } from '../Input';
import './DatePicker.scss';

export default class DatePicker extends Component {
  static defaultProps = {
    currentMonth: new Date().getMonth(),
    currentYear: new Date().getFullYear(),
    selectedDate: null
  };

  static propTypes = {
    selectedDate: PropTypes.oneOfType([() => null, PropTypes.instanceOf(Date)]),
    currentMonth: PropTypes.number,
    currentYear: PropTypes.number
  };

  state = {
    today: {
      day: new Date().getDate(),
      month: new Date().getMonth(),
      year: new Date().getFullYear()
    },
    selectedDate: this.props.selectedDate
      ? getSimpleDate(this.props.selectedDate)
      : null,
    currentMonth:
      this.props.currentMonth ||
      (this.props.selectedDate
        ? this.props.selectedDate.getMonth()
        : new Date().getMonth()),
    currentYear:
      this.props.currentYear ||
      (this.props.selectedDate
        ? this.props.selectedDate.getFullYear()
        : new Date().getFullYear()),
    dateString: toSimpleDateString(this.props.selectedDate) || ''
  };

  resetCalendar = () => {
    const { currentMonth, currentYear, selectedDate } = this.props;
    this.setState({
      selectedDate,
      currentMonth:
        currentMonth ||
        (selectedDate ? selectedDate.getMonth() : new Date().getMonth()),
      currentYear:
        currentYear ||
        (selectedDate ? selectedDate.getFullYear() : new Date().getFullYear())
    });
  };

  selectDate = date => {
    if (typeof date === 'string') {
      const newDate = toSimpleDateObj(date);
      const { month, year } = newDate;
      const { currentMonth, currentYear } = this.state;
      this.setState({
        selectedDate: newDate
      });
      if (month !== currentMonth || year !== currentYear) {
        this.setCurrentMonth({ month, year });
      }
    } else {
      this.setState({
        selectedDate: date,
        dateString: toSimpleDateString(date)
      });
    }
  };

  setCurrentMonth = ({ month, year }) => {
    this.setState({
      currentMonth: month,
      currentYear: year
    });
  };

  setCurrentYear = year => {
    this.setState({
      currentYear: year
    });
  };

  updateDateString = e => {
    const { value } = e.target;
    this.setState({
      dateString: value
    });
    if (isSDSFormat(value)) {
      this.selectDate(value);
    }
  };

  setDate = () => {
    const { selectedDate: currentDueDate, selectDate } = this.props;
    const { selectedDate } = this.state;
    if (
      !(currentDueDate === null && selectedDate === null) &&
      !isSameDate(currentDueDate, selectedDate)
    ) {
      const { day, month, year } = selectedDate;
      const date = new Date(year, month, day);
      selectDate(date);
    }
    this.closeDatePicker();
  };

  clearDueDate = () => {
    const { selectedDate, selectDate } = this.props;
    this.selectDate(null);
    if (selectedDate !== null) {
      selectDate(null);
    }
    this.closeDatePicker();
  };

  closeDatePicker = () => {
    const { onClose } = this.props;
    if (onClose) {
      onClose();
    }
  };

  cancel = () => {
    this.closeDatePicker();
    this.resetCalendar();
  };

  render() {
    const { isActive, innerRef } = this.props;
    const {
      today,
      selectedDate,
      currentMonth,
      currentYear,
      dateString
    } = this.state;
    return (
      <div
        className="date-picker"
        ref={innerRef}
        style={{ display: !isActive ? 'none' : 'block' }}
      >
        <div className="date-picker__header">
          <div className="date-picker__due-date-wrapper">
            <Input
              name="dueDate"
              type="text"
              label="Due Date"
              value={dateString}
              onChange={this.updateDateString}
              className="date-picker__input--due-date"
              labelClass="date-picker__label--due-date"
              placeholder="mm-dd-yy"
              maxLength={8}
            />
          </div>
        </div>
        <Calendar
          classes={{
            calendar: 'date-picker__calendar',
            weekday: 'date-picker__week-day'
          }}
          month={currentMonth}
          year={currentYear}
          selectedDate={selectedDate}
          today={today}
          onDayClick={this.selectDate}
          onMonthClick={this.setCurrentMonth}
          onYearClick={this.setCurrentYear}
        />
        <div className="date-picker__footer">
          <Button
            size="sm"
            variant="text"
            color="neutral"
            className="date-picker__btn--clear"
            onClick={this.clearDueDate}
          >
            Clear
          </Button>
          <Button
            size="sm"
            variant="contained"
            color="primary"
            className="date-picker__btn--set-due-date"
            onClick={this.setDate}
          >
            Done
          </Button>
          <Button
            size="sm"
            variant="text"
            color="neutral"
            className="date-picker__btn--cancel"
            onClick={this.cancel}
          >
            Cancel
          </Button>
        </div>
      </div>
    );
  }
}
