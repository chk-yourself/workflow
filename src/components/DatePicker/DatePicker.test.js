import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import DatePicker from './DatePicker';

describe('Date Picker Component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<DatePicker />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with default currentMonth value', () => {
    const component = mount(<DatePicker />);
    expect(component.prop('currentMonth').toEqual(new Date().getMonth()));
  });
});
