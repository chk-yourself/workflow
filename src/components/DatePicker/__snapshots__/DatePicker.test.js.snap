// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DatePicker /> renders correctly 1`] = `
<div
  className="date-picker"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="date-picker__header"
  >
    <div
      className="date-picker__due-date-wrapper"
    >
      <label
        className="input__label date-picker__label--due-date "
        htmlFor="dueDate"
      >
        Due Date
      </label>
      <input
        className="input date-picker__input--due-date"
        id="dueDate"
        maxLength={8}
        name="dueDate"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="mm-dd-yy"
        readOnly={false}
        required={false}
        tabIndex={0}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="calendar date-picker__calendar"
  >
    <div
      className="calendar__year"
    >
      <div
        className="popover-wrapper  calendar__years-dropdown-wrapper"
        style={null}
      >
        <button
          className="calendar__btn--years-dropdown  text text--neutral btn--md "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          2019
          <svg
            className="feather feather-chevron-down "
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <use
              xlinkHref="feather-sprite.svg#chevron-down"
            />
          </svg>
        </button>
        <div
          className="popover align-left calendar__years-dropdown"
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
        >
          <ul
            className="calendar__years-list"
          >
            <li
              className="calendar__years-item"
            >
              <input
                checked={true}
                className="radio calendar__radio"
                id={2019}
                name="year"
                onChange={[Function]}
                type="radio"
                value={2019}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor={2019}
              >
                <svg
                  className="feather feather-check "
                  fill="none"
                  height={24}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <use
                    xlinkHref="feather-sprite.svg#check"
                  />
                </svg>
                2019
              </label>
            </li>
            <li
              className="calendar__years-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id={2020}
                name="year"
                onChange={[Function]}
                type="radio"
                value={2020}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor={2020}
              >
                2020
              </label>
            </li>
            <li
              className="calendar__years-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id={2021}
                name="year"
                onChange={[Function]}
                type="radio"
                value={2021}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor={2021}
              >
                2021
              </label>
            </li>
            <li
              className="calendar__years-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id={2022}
                name="year"
                onChange={[Function]}
                type="radio"
                value={2022}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor={2022}
              >
                2022
              </label>
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="calendar__month"
    >
      <div
        className="popover-wrapper  calendar__months-dropdown-wrapper"
        style={null}
      >
        <button
          className="calendar__btn--months-dropdown  text text--neutral btn--md "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          June
          <svg
            className="feather feather-chevron-down "
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <use
              xlinkHref="feather-sprite.svg#chevron-down"
            />
          </svg>
        </button>
        <div
          className="popover align-left calendar__months-dropdown"
          onClick={[Function]}
          style={
            Object {
              "display": "none",
            }
          }
        >
          <ul
            className="calendar__months-list"
          >
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="January"
                name="month"
                onChange={[Function]}
                type="radio"
                value={0}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="January"
              >
                January
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="February"
                name="month"
                onChange={[Function]}
                type="radio"
                value={1}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="February"
              >
                February
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="March"
                name="month"
                onChange={[Function]}
                type="radio"
                value={2}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="March"
              >
                March
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="April"
                name="month"
                onChange={[Function]}
                type="radio"
                value={3}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="April"
              >
                April
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="May"
                name="month"
                onChange={[Function]}
                type="radio"
                value={4}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="May"
              >
                May
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={true}
                className="radio calendar__radio"
                id="June"
                name="month"
                onChange={[Function]}
                type="radio"
                value={5}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="June"
              >
                <svg
                  className="feather feather-check "
                  fill="none"
                  height={24}
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 24 24"
                  width={24}
                  xmlns="http://www.w3.org/2000/svg"
                  xmlnsXlink="http://www.w3.org/1999/xlink"
                >
                  <use
                    xlinkHref="feather-sprite.svg#check"
                  />
                </svg>
                June
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="July"
                name="month"
                onChange={[Function]}
                type="radio"
                value={6}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="July"
              >
                July
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="August"
                name="month"
                onChange={[Function]}
                type="radio"
                value={7}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="August"
              >
                August
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="September"
                name="month"
                onChange={[Function]}
                type="radio"
                value={8}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="September"
              >
                September
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="October"
                name="month"
                onChange={[Function]}
                type="radio"
                value={9}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="October"
              >
                October
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="November"
                name="month"
                onChange={[Function]}
                type="radio"
                value={10}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="November"
              >
                November
              </label>
            </li>
            <li
              className="calendar__months-item"
            >
              <input
                checked={false}
                className="radio calendar__radio"
                id="December"
                name="month"
                onChange={[Function]}
                type="radio"
                value={11}
              />
              <label
                className="radio__label calendar__radio-label"
                htmlFor="December"
              >
                December
              </label>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="calendar__month--prev-next"
      >
        <button
          className="calendar__btn--month calendar__btn--prev-month text text--neutral icon-only icon-only--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          <svg
            className="feather feather-chevron-left "
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <use
              xlinkHref="feather-sprite.svg#chevron-left"
            />
          </svg>
        </button>
        <button
          className="calendar__btn--month calendar__btn--next-month text text--neutral icon-only icon-only--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          <svg
            className="feather feather-chevron-right "
            fill="none"
            height={24}
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 24 24"
            width={24}
            xmlns="http://www.w3.org/2000/svg"
            xmlnsXlink="http://www.w3.org/1999/xlink"
          >
            <use
              xlinkHref="feather-sprite.svg#chevron-right"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      className="calendar__week-days"
    >
      <div
        className="calendar__week-day date-picker__week-day"
      >
        S
      </div>
      <div
        className="calendar__week-day date-picker__week-day"
      >
        M
      </div>
      <div
        className="calendar__week-day date-picker__week-day"
      >
        T
      </div>
      <div
        className="calendar__week-day date-picker__week-day"
      >
        W
      </div>
      <div
        className="calendar__week-day date-picker__week-day"
      >
        T
      </div>
      <div
        className="calendar__week-day date-picker__week-day"
      >
        F
      </div>
      <div
        className="calendar__week-day date-picker__week-day"
      >
        S
      </div>
    </div>
    <div
      className="calendar__days"
    >
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          26
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          27
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          28
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          29
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          30
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          31
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          1
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          2
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          3
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          4
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          5
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          6
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          7
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          8
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          9
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          10
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          11
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          12
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          13
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          14
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          15
        </button>
      </div>
      <div
        className="calendar__day  is-today  undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          16
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          17
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          18
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          19
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          20
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          21
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          22
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          23
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          24
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          25
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          26
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          27
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          28
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          29
        </button>
      </div>
      <div
        className="calendar__day    undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          30
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          1
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          2
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          3
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          4
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          5
        </button>
      </div>
      <div
        className="calendar__day calendar__day--prev-next   undefined"
      >
        <button
          className="calendar__btn--day text text--neutral btn--sm "
          name=""
          onClick={[Function]}
          onMouseDown={[Function]}
          type="button"
          value=""
        >
          6
        </button>
      </div>
    </div>
  </div>
  <div
    className="date-picker__footer"
  >
    <button
      className="date-picker__btn--clear text text--neutral btn--sm "
      name=""
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
      value=""
    >
      Clear
    </button>
    <button
      className="date-picker__btn--set-due-date contained contained--primary btn--sm "
      name=""
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
      value=""
    >
      Done
    </button>
    <button
      className="date-picker__btn--cancel text text--neutral btn--sm "
      name=""
      onClick={[Function]}
      onMouseDown={[Function]}
      type="button"
      value=""
    >
      Cancel
    </button>
  </div>
</div>
`;
